src/seedu/addressbook/commands/DeleteCommand.java

```diff
@@ -11,27 +14,36 @@
 public class DeleteCommand extends Command {
 
     public static final String COMMAND_WORD = "delete";
-
+    
     public static final String MESSAGE_USAGE = COMMAND_WORD
             + ": Deletes the person identified by the index number used in the last person listing.\n"
             + "Parameters: INDEX\n"
             + "Example: " + COMMAND_WORD + " 1";
 
     public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
+    public static final String MESSAGE_DELETE_PERSON_CANCELLED = "Deletion Cancelled";
+
+    public static final String MESSAGE_DELETE_CONFIRM = "|| Do you really want to delete this entry? " +
+            "Press 1 to confirm, and 0 to cancel: ";
 
 
-    public DeleteCommand(int targetVisibleIndex) {
+    public DeleteCommand(int targetVisibleIndex, TextUi ui) {
         super(targetVisibleIndex);
+        this.ui = ui;
     }
 
-
     @Override
     public CommandResult execute() {
         try {
             final ReadOnlyPerson target = getTargetPerson();
-            addressBook.removePerson(target);
-            return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, target));
-
+            System.out.println(MESSAGE_DELETE_CONFIRM);
+            int answer = ui.getUserConfirmation();
+            if(answer == 1){
+                addressBook.removePerson(target);
+                return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, target));
+            }else{
```

Comment:
[Coding standard](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html) violation! One more more of these problems detected:
1. extra/missing spaces.
2. brace style problem.
3. missing braces.
4. name doesn't follow naming convention